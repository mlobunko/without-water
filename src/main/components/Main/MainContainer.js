import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { MainComponent } from './MainComponent';
import ErrorBoundary from '../ErrorBoundary';

import authorized from 'shared/authorized';

export class MainContainer extends React.Component {
  render() {
    return (
      <ErrorBoundary>
        <MainComponent authorized={this.props.authorized} />
      </ErrorBoundary>
    );
  }
}

const mapStateToProps = state => ({
  authorized: authorized.selectors.getAll(state)
});

export default withRouter(connect(mapStateToProps)(MainContainer));
