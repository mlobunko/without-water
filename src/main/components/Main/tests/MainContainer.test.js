import React from 'react';
import { shallow } from 'enzyme';

import { MainContainer } from '../MainContainer';

describe('MainContainer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<MainContainer authorized={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
