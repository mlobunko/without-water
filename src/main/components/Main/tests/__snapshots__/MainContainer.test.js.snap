// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer renders correctly 1`] = `
<ErrorBoundary>
  <MainComponent
    authorized={true}
  />
</ErrorBoundary>
`;
